src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -140,6 +140,8 @@ public CredentialPromptWebRequestHandler(IWebProxy proxy)
                                 var credentials = await HttpHandlerResourceV3
                                     .PromptForProxyCredentials(request.RequestUri, Proxy, cancellationToken);
 
+                                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Why? I would like to avoid peppering `.ThrowIfCancellationRequested()` all over the place unless there is a major perf benefit or if it fixes a bug (as it does in this case).